clippy_lints/src/literal_representation.rs

```diff
@@ -529,19 +529,23 @@ impl LiteralRepresentation {
             then {
                 let digit_info = DigitInfo::new(&src, false);
                 if digit_info.radix == Radix::Decimal {
-                    let val = digit_info.digits
+                    if let Ok(val) = digit_info.digits
                         .chars()
                         .filter(|&c| c != '_')
                         .collect::<String>()
-                        .parse::<u128>().unwrap();
-                    if val < u128::from(self.threshold) {
-                        return
+                        .parse::<u128>() {
+                        if val < u128::from(self.threshold) {
+                            return
+                        }
+                        let hex = format!("{:#X}", val);
+                        let digit_info = DigitInfo::new(&hex[..], false);
+                        let _ = Self::do_lint(digit_info.digits).map_err(|warning_type| {
+                            warning_type.display(&digit_info.grouping_hint(), cx, lit.span)
+                        });
                     }
-                    let hex = format!("{:#X}", val);
-                    let digit_info = DigitInfo::new(&hex[..], false);
-                    let _ = Self::do_lint(digit_info.digits).map_err(|warning_type| {
-                        warning_type.display(&digit_info.grouping_hint(), cx, lit.span)
-                    });
+                    else {
+                        return
```

Comment:
NIT: `else` formatting

Is this `else { return }` required? It's not in a loop and the function ends after the `if_chain!` macro.

Reply:
Good catch, no it's not required =)